src/settings.ts

```diff
@@ -139,6 +147,24 @@ export async function parseSettings(
     },
   );
 
+  const logger = await check(
+    "logger",
+    (logger = console): MigrateLogger => {
+      const typedLogger = logger as MigrateLogger;
+      if (
+        typeof logger !== "object" ||
+        !("log" in typedLogger && typeof typedLogger.log === "function") ||
```

Comment:
I was not really sure if such validation is needed here since typescript should anyway give you errors if provided logger object does not match the `MigrateLogger` interface... but decided to be safe.

Reply:
Outdated, now only checking if the provided factory is a function.